packages/11ty/_lib/urlPathJoin/index.js

```diff
@@ -0,0 +1,23 @@
+import { posix as path } from 'node:path'
+/**
+ * @function urlPathJoin
+ *
+ * @argument {String} url
+ * @argument {Iterable} segs
+ *
+ * Parses `url` and safely joins all of `segs` to its pathname
+ *
+ * @return {String}
+ *
+ **/
+
+const urlPathJoin = (url, ...segs) => {
+  const base = new URL(url)
+  const basePath = base.pathname
+
+  base.pathname = path.join(basePath, ...segs)
+
+  return base.href
+}
```

Comment:
⚙️ I think the following is a more idiomatic use of the node api
```suggestion
const urlPathJoin = (baseUrl, ...segments) => {
  const url = new URL(path.join(...segments), baseUrl)
  return url.toString()
}
```